src/App.tsx

```diff
@@ -14,11 +15,115 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+interface State {
+  goods: string[];
+  selectedGood: string[];
+}
 
-export default App;
+export class App extends React.Component<{}, State> {
+  state = {
+    goods: [...goodsFromServer],
+    selectedGood: [''],
+  };
+
+  oldSelected = [''];
+
+  addButton = (good: string) => {
+    this.oldSelected = [...this.state.selectedGood];
+    this.setState({
+      selectedGood: [...this.oldSelected, good],
+    });
```

Comment:
Use functional `setState` here
```suggestion
    this.setState(state => ({
        selectedGoods: [...state.selectedGoods, good],
      }));
```